### PR TITLE
fix(ui): two shortcut corrections the admin migration exposed

### PR DESCRIPTION
**Pull request by mobeenabdullah**
Follow-up to #612, for a Codex finding that arrived four minutes before it merged.

## The defect

A held key's suppression was re-decided by comparing a signature of the things that might have changed: the modifier flags, and a string describing the layer stack. A binding's own `when` was not in that signature, and **could not be** — it is a function, so every render supplies a new identity while the condition is unchanged, and signing it would re-decide constantly.

So a binding whose action falsifies its own condition kept its first decision for as long as the key was held. Codex's case: a blocking modal's `Space` binding with `preventDefault: false` sets its own condition false on the first press. The repeats should then fall through to the blocking layer and be suppressed. Instead the page went on scrolling underneath a modal that claims to hold the keyboard.

The comment on the old code said `when` "is re-read at press time instead". That was true of a fresh press and false of the repeat fast path — which is the path this bug takes.

## The fix

Ask instead of compare. A permitted press is re-offered to the stack on each repeat, with `invoke` false so the matching binding's `preventDefault` policy applies without its action running again.

That reads all three inputs at once — modifiers, layer stack, and `when` — so it subsumes both signature comparisons rather than adding a third clause beside them. A *prevented* press stays sticky, unchanged: adding shift to a held `mod+s` must not hand the browser a Save As.

## What this removes

The signature machinery had no other consumer, so it goes: `stackSignature()`, `layerPolicy()`, the `policy` field on a registered layer, `modifierState()`, and the `modifiers` and `stack` fields on each consumed press. `manager.ts` is about 50 lines shorter.

Leaving it in place would have been worse than dead weight — it would tell the next reader that suppression is decided by signatures, which is exactly the belief that produced this bug.

## Tests

Two new cases, plus the existing 171 unchanged.

- **The regression.** The binding lives IN the blocking layer, which is already up before the key goes down, so the stack never changes and the modifiers never move. Only `when` changes.
- **The control.** With no blocking layer above it, a binding that still matches and still sets `preventDefault: false` leaves its repeats to the browser — re-deciding must not become "suppress every repeat", or a held key stops being usable for panning or scrolling.

### A note on how the test was arrived at

My first version of the regression test **passed against the unfixed code**. Its handler *registered* the modal, which changed the layer stack, so the old signature comparison caught it for an unrelated reason and the test proved nothing. Break-verification is what surfaced that.

The committed version puts the blocking layer up before the press, which is the only arrangement that isolates `when`. Restoring the old signature comparison now fails that test and nothing else (1 failed | 172 passed).

## Verification

596 tests in `packages/ui`, `lint` 0, `check-types` 0, `build` clean. The 171 pre-existing shortcut tests pass unchanged, including the two that previously covered the modifier-change and stack-change paths — which is the evidence that re-offering subsumes both.

### CODE REVIEW

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/ui/src/lib/shortcuts/manager.ts [hunk 1]
```diff
@@ -876,17 +824,25 @@ export function createShortcutManager(
...
           event.preventDefault();
-        } else if (
-          held.modifiers !== modifierState(event) ||
-          held.stack !== stackSignature()
-        ) {
-          if (!insertsText(event, typing)) event.preventDefault();
+        } else {
+          const still = offer([event], event, typing, false);
+          if (still === "blocked" && !insertsText(event, typing)) {
+            event.preventDefault();
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/ui/src/lib/shortcuts/manager.ts:852]: Confirmed and fixed in `3e04cfab6`. A real regression I introduced in this PR, and you were right about the mechanism.

The re-offer now suppresses on any outcome that is not `"fired"`, not only on `"blocked"`. `"fired"` means a binding matched and its own `preventDefault` policy has already been applied, so nothing more is decided; every other outcome leaves a press this manager still owns with no one applying a policy to it, and the browser must not act on a key we claim — unless it is text, which is the one thing a grab may never eat.

**Why the existing coverage missed it, which is the part worth recording.** "a permitted key whose modifiers change mid-hold" types into an input UNDER a blocking modal, so its re-offer returns `"blocked"` and it passed against the broken code. It never exercised the `"none"` path at all.

My first attempt at the regression test failed against the FIX as well: I copied that test's setup and dispatched a bare `w` from an input, where a bare character binding does not fire while typing, so the first press was never consumed and the repeat took a different branch entirely. The committed version dispatches away from a field, where the press is genuinely claimed, and carries a control asserting the first press is still permitted.

Verified by break: narrowing the condition back to `still === "blocked"` fails that test alone (1 failed | 174 passed). 598 tests in `packages/ui`, lint and types clean.

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/ui/src/lib/shortcuts/react.tsx [hunk 2]
```diff
@@ -234,20 +234,18 @@ export function ShortcutProvider({
       "options, so the ones passed here are being ignored. Managers are shared per target; give " +
       "the providers matching options, or a target of their own."
   );
+  // Nesting with matching options is SUPPORTED, not a mistake, and is deliberately not warned
+  // about: a component that owns keys and can be rendered on its own — a command palette exported
+  // for embedding — has to bring a provider with it, or it throws wherever no shell wrapped it.
+  // Reusing the target's manager makes that composition free, and the warning above still reports
+  // the case that genuinely loses something, which is options that disagree.
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/ui/src/lib/shortcuts/react.tsx:241]: Confirmed, and fixed in **#658** — this landed as #654 merged, so it goes into a follow-up rather than being lost here.

You are right, and it is a regression from my own change in this PR. I removed the generic nesting warning on the argument that the adjacent options-conflict warning already covers the case that loses something. That holds for providers sharing a DOM target and NOT for detached ones: `ownersByTarget` is a WeakMap keyed by the target, `null` is not a key, so `target={null}` has no registry entry to compare against and the warning could never fire for it.

Not a corner case, either — the admin's shared test render uses `target={null}` precisely so no listener touches the shared jsdom document.

#658 carries the options fingerprint through context and checks the PARENT as well as the registry: the registry answers for a shared target, the parent answers for the nested case including the detached one. `ShortcutScope` inherits the fingerprint rather than producing one, since a scope raises precedence without building a manager.

Tests: two nested detached providers with conflicting `isApple` warn; two that agree stay silent, which is the composition #654 deliberately made free. Break-verified by narrowing back to the registry alone.
